src/main/java/marquez/api/models/JobResponse.java

```diff
@@ -14,26 +14,43 @@
 
 package marquez.api.models;
 
+import static marquez.common.base.MorePreconditions.checkNotBlank;
+import static marquez.common.base.MorePreconditions.checkNotEmpty;
+
 import java.util.List;
-import lombok.AllArgsConstructor;
-import lombok.Data;
-import lombok.NoArgsConstructor;
+import java.util.Optional;
+import javax.annotation.Nullable;
+import lombok.*;
 
-@Data
-@AllArgsConstructor
-@NoArgsConstructor
+@EqualsAndHashCode
+@ToString
 public final class JobResponse {
-  private String name;
-
-  private String createdAt;
-
-  private String updatedAt;
-
-  private List<String> inputDatasetUrns;
-
-  private List<String> outputDatasetUrns;
-
-  private String location;
-
+  @Getter private String name;
+  @Getter private String createdAt;
+  @Getter private String updatedAt;
+  @Getter private List<String> inputDatasetUrns;
+  @Getter private List<String> outputDatasetUrns;
+  @Getter private String location;
   private String description;
+
+  public JobResponse(
+      @NonNull final String name,
+      @NonNull final String createdAt,
+      @NonNull final String updatedAt,
+      @NonNull final List<String> inputDatasetUrns,
+      @NonNull final List<String> outputDatasetUrns,
+      @NonNull String location,
+      @Nullable String description) {
+    this.name = checkNotBlank(name);
+    this.createdAt = checkNotBlank(createdAt);
+    this.updatedAt = checkNotBlank(updatedAt);
+    this.inputDatasetUrns = (List<String>) checkNotEmpty(inputDatasetUrns);
```

Comment:
Both `JobResponse.inputDatasetUrns` and `JobResponse.outputDatasetUrns` may be empty. That is, it’s possible for a job to have only input, or only output datasets defined (see: https://marquezproject.github.io/marquez/#data-model)

Let's remove the assertion `MorePreconditions.checkNotEmpty()`. We can also achieve the same check with [`Preconditions.checkArgument()`](https://guava.dev/releases/snapshot/api/docs/com/google/common/base/Preconditions.html#checkArgument-boolean-). For example,

```java
void print(List<String> lines) {
  Preconditions.checkArgument(!lines.isEmpty, "lines must not be empty")
  ...
}
```

Reply:
sure

Reply:
done